plugins/redis.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const Redis = require('ioredis');
+const logger = require('screwdriver-logger');
+const { connectionDetails } = require('../config/redis');
+
+const redis = new Redis({
+    port: connectionDetails.port,
+    host: connectionDetails.host,
+    password: connectionDetails.options.password,
+    tls: connectionDetails.options.tls
+});
+
+redis.on('connecting', () => {
+    logger.info('Connecting to Redis.');
+});
+redis.on('connect', () => {
+    logger.info('Successfully connected to Redis');
+});
+redis.on('error', err => {
+    if (err.code === 'ECONNREFUSED') {
+        logger.error(`Could not connect to Redis: ${err.message}.`);
```

Comment:
What would be the implication of this in the queue service?
Can queue service operate without connecting to redis?

Reply:
ioredis will retry until `maxRetriesPerRequest`, we just catching this to log the error. Behavior will be same regardless

Reply:
How are we handling the cases of periodic job configurations getting lost?
Will they get fixed if the redis error is resolved?